### ISSUE TITLE
Run in Background

### ISSUE THREAD

**Issue by Studiodews** [amitshekhariitbhu/Fast-Android-Networking#534]
Please I want to know if request sent run in Background?

**Comment by amitshekhariitbhu** [comment]
Hi @Studiodews 

Yes.

**Comment by Studiodews** [comment]
@amitshekhariitbhu any requests at all? 😃.

**Comment by amitshekhariitbhu** [comment]
There are two types of request in library:
* asynchronous: runs on background thread
* synchronous: runs on thread from where we call it.

**Comment by Studiodews** [comment]
@amitshekhariitbhu thank you, can I get a tutorial or example link on that from you?

**Comment by amitshekhariitbhu** [comment]
@Studiodews welcome
Async: [Link](https://amitshekhar.me/Fast-Android-Networking/get_request.html)
Sync: [Link](https://amitshekhar.me/Fast-Android-Networking/synchronous_request.html)

**Comment by Studiodews** [comment]
Thank you

**Comment by Studiodews** [comment]
I found out something, the difference in the Aysnc and Sync tasks is that 
Async :AndroidNetworking.get("https://fierce-cove-29863.herokuapp.com/getAnUserDetail/{userId}")

Sync : ANRequest request = AndroidNetworking

So here Sync task is as if it is creating an instance of a of a class and I can't differentiate whether the class is ANRequest or AndroidNetworking. So my question is if I should extend your library, first the Aysnc task, do you think I should create an instance (instancemyclass = new My class.get(url)) of my class which extended yours or I should just call it "Myclass.get(url)" ?

Then my second question if I should extend Synchronous tasks to what do I extend it?

So in summary I want to know in extending your class how do I do it?

**Comment by amitshekhariitbhu** [comment]
I will recommend to check AndroidNetworking and ANRequest classes. If you want to create a wrapper class, just create over the AndroidNetworking class.

**Comment by Studiodews** [comment]
Okay, thank you
